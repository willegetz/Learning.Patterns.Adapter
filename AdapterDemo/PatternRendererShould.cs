using AdapterDemo.Model;
using ApprovalTests;
using ApprovalTests.Reporters;
using Microsoft.VisualStudio.TestTools.UnitTesting;
using System.Collections.Generic;

namespace AdapterDemo
{
    [TestClass]
    [UseReporter(typeof(VisualStudioReporter))]
    public class PatternRendererShould
    {
        /*
         * We want the PatternRenderer to display its data like the DataRenderer displays data.
         * Before we can refactor, we have to finish writing our tests to ensure the code doesn't
         * break as we work.
         */

        [TestMethod]
        public void RenderOneRowGivenOnePatternInList()
        {
            var pattern = new Pattern
            {
                Id = 1,
                Name = "A Pattern",
                Description = "Test object for testing."
            };

            var patterns = new List<Pattern>() { pattern };

            var renderer = new PatternRenderer();

            var result = renderer.ListPatterns(patterns);

            Approvals.Verify(result);
        }

        [TestMethod]
        public void RenderTwoRowsGivenTwoPatternsInList()
        {
            var patternOne = new Pattern
            {
                Id = 1,
                Name = "A Pattern",
                Description = "Test object for testing."
            };

            var patternTwo = new Pattern
            {
                Id = 2,
                Name = "Another Pattern",
                Description = "A friend to pattern one."
            };

            var patterns = new List<Pattern>() { patternOne, patternTwo };

            var renderer = new PatternRenderer();

            var result = renderer.ListPatterns(patterns);

            Approvals.Verify(result);
        }
    }
}