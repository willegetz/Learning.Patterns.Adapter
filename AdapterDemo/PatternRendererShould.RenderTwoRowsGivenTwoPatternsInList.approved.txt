Rendering Data:
Id                   Name                 Description          
1                    A Pattern            Test object for testing. 
2                    Another Pattern      A friend to pattern one.